### PR TITLE
[storm-elasticsearch] Expose TransportClient configuration Map to EsConfig

### PR DESCRIPTION
**Pull request by alexpanov**

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
@alexpanov 
Nice work. I have some comments for this PR,

1. 
I found you're refactoring whole codes which I think is better to file several new issues to JIRA respectively. 
Please keep diffs simple and small enough, and try to avoid introducing unrelated diffs.

2. 
Seems like you need to update your working directory before starting.
I merged #679 into master after merging #693, which was conflicted.

3. storm-core relocates guava's package for avoiding conflicts. 
I'm not 100% sure, but I think it means that if we need to use guava from outer of storm-core, guava should be added as dependency explicitly.
Any other external modules already did it, though I think it would be better to relocate guava again.

**Comment by alexpanov** [comment]
@HeartSaVioR 
1. No problem. Will do.
2. Done.
3. I used ES Preconditions instead. It's already in the classpath anyway.

I also updated documentation for the EsConfig (since I have made it immutable previously).

**Comment by harshach** [comment]
@alexpanov can you also fix the PR title to JIRA title

**Comment by alexpanov** [comment]
@harshach Done

File: external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/SniffSettings.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.apache.storm.elasticsearch.common;
```
Comment (HeartSaVioR) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/SniffSettings.java]: @alexpanov Seems like it is unused class. Could you remove this class?

File: external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/EsConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package org.apache.storm.elasticsearch.common;
```
Comment (HeartSaVioR) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/EsConfigTest.java]: @alexpanov Please attach Apache License comment above here.

File: external/storm-elasticsearch/README.md [hunk 3]
```diff
@@ -84,6 +84,7 @@ Provided components (Bolt, State) takes in EsConfig as a constructor arg.
 |---	|--- |---
 |clusterName | Elasticsearch cluster name | String (required) |
 |nodes | Elasticsearch nodes in a String array, each element should follow {host}:{port} pattern | String array (required) |
+|additionalParameters | Additional Elasticsearch configuration parameters | Map<String, String> (optional) |
```
Comment (HeartSaVioR) [inline external/storm-elasticsearch/README.md]: @alexpanov It would be better to describe that it will be applied to TransportClient.

**Comment by HeartSaVioR** [comment]
Overall looks fine. I left some comments, most of things are about license comment.

File: external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/SniffSettings.java (hunk 1 above)
Comment (alexpanov) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/SniffSettings.java]: Done. Sorry, that's an experimental leftover. My bad.

File: external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/EsConfigTest.java (hunk 2 above)
Comment (alexpanov) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/common/EsConfigTest.java]: Done

File: external/storm-elasticsearch/README.md (hunk 3 above)
Comment (alexpanov) [inline external/storm-elasticsearch/README.md]: Done

File: external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/trident/EsStateFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package org.apache.storm.elasticsearch.trident;
```
Comment (HeartSaVioR) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/trident/EsStateFactoryTest.java]: @alexpanov Sorry I missed. Please attach Apache License comment above here.

Comment (alexpanov) [inline external/storm-elasticsearch/src/test/java/org/apache/storm/elasticsearch/trident/EsStateFactoryTest.java]: @HeartSaVioR Done

**Comment by HeartSaVioR** [comment]
@alexpanov 
Please change title to include STORM-1013 as prefix, and JIRA title - refer #698 for example.

**Comment by HeartSaVioR** [comment]
@alexpanov 
I merged into local, and ran ```mvn clean install```, and get some errors.

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/heartsavior/WorkArea/Playground/storm/external/storm-elasticsearch/src/main/java/org/apache/storm/elasticsearch/common/EsConfig.java:[24,33] package com.google.common.collect does not exist
[ERROR] /Users/heartsavior/WorkArea/Playground/storm/external/storm-elasticsearch/src/main/java/org/apache/storm/elasticsearch/common/EsConfig.java:[24,33] package com.google.common.collect does not exist
[ERROR] /Users/heartsavior/WorkArea/Playground/storm/external/storm-elasticsearch/src/main/java/org/apache/storm/elasticsearch/common/EsConfig.java:[70,40] cannot find symbol
  symbol:   variable ImmutableMap
  location: class org.apache.storm.elasticsearch.common.EsConfig
```

Maybe usages of guava is not removed clearly.

**Comment by alexpanov** [comment]
@HeartSaVioR My bad. I should've run the check. Anyway, it is done.

**Comment by HeartSaVioR** [comment]
Great. +1.

**Comment by HeartSaVioR** [comment]
Thanks @alexpanov, I merged with master.
But github mirror is a bit behind. I'll mark STORM-1013 as resolved when mirror catches up.

**Pull request merged**